### PR TITLE
Modifies the user ID cookie to have a configurable TTL

### LINKED ISSUE #326: Make Cookie TTLs configurable
Right now, the TTL on synced User IDs is hardcoded: https://github.com/prebid/prebid-server/blob/master/pbs/usersync.go#L146

For some reason, we need to set a different TTL (I didn't get the details, but @mkendall07 has them)... so we plan to make this configurable.

### PR DESCRIPTION
**Pull request by hhhjort**
Fixes #326

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ type HostCookie struct {
 	OptOutURL    string `mapstructure:"opt_out_url"`
 	OptInURL     string `mapstructure:"opt_in_url"`
 	OptOutCookie Cookie `mapstructure:"optout_cookie"`
+	// Cookie timeout in days
+	TTL int64 `mapstructure:"ttl"`
```
Comment (dbemiller) [inline config/config.go]: Can we call this `ttl_days`? Less chance of screwing it up while editing the config.

File: pbs_light.go [hunk 2]
```diff
@@ -885,6 +885,7 @@ func serve(cfg *config.Configuration) error {
 		OptOutURL:    cfg.HostCookie.OptOutURL,
 		OptInURL:     cfg.HostCookie.OptInURL,
 		OptOutCookie: cfg.HostCookie.OptOutCookie,
+		TTL:          time.Duration(cfg.HostCookie.TTL) * 24 * time.Hour,
```
Comment (dbemiller) [inline pbs_light.go]: Are you sure this is right?

It looks like you're multiplying by `* 24 * time.Hour` both here and in `ToHTTPCookie` in `usersync.go`. I'm not certain, but... seems like a bug?

File: config/config.go (hunk 1 above)
Comment (hhhjort) [reply config/config.go]: Done

File: pbs_light.go (hunk 2 above)
Comment (hhhjort) [reply pbs_light.go]: Good catch. go is fine implicitly converting a static `int` to `time.Duration`, but not converting a variable of type `int64`. So I found I had to cascade back a `time.Duration`declaration, and at the end (beginning) I again did a day-time conversion. I think its cleaner to do the conversion at the front, and let `ttl` be a `time.duration` throughout the code than do it as the last step.

Comment (dbemiller) [reply pbs_light.go]: I had almost exactly the same bug recently. I don't like how Go implements these type aliases at all.

I think your strategy is the right one, though. Convert once ASAP, and pass the value through from there.

**Comment by hhhjort** [comment]
I added a default for the TTL. Did notice that we don't have defaults for any of the other cookie parameters, but on second look I don't think we can pick sensible defaults for the remainder. This will keep that app backwards compatible with old configs, which is plenty of reason to do it still.

**Comment by mkendall07** [comment]
@bretg fyi on this - we added a config for pbs cookie ttl but it should be backwards compatible.

**Pull request merged**